### PR TITLE
Add Sudo Commands Bundle 

### PR DESCRIPTION
**Pull request by UlfatSayyed-okta**
CReate a new provider for sudo commands bundle. 
Used https://github.com/atko-pam/pam-sdk-go/pull/64 to generate the client

### CODE REVIEW

File: oktapam/resource_sudo_commands_bundle.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	if err := client.UpdateSudoCommandsBundle(ctx, c, sudoCommandBundle); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(*sudoCommandBundle.Id)
+	return resourceSudoCommandsBundleRead(ctx, d, m)
+}
+
+func resourceSudoCommandsBundleDelete(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	// there isn't a true delete with password settings, and as databases don't specify ManagedPrivilegedAccounts,
```
Comment (SajanAlexander-okta) [inline oktapam/resource_sudo_commands_bundle.go:219]: is this comment relevant?

File: oktapam/resource_sudo_commands_bundle_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	scb, err := client.GetSudoCommandsBundle(context.Background(), c, id)
+	return scb != nil && *scb.Id != "" && err == nil, err
+}
+
+func createTestAccSudoCommandsBundleCreateConfig(name string) string {
+	const format = `
+	resource "oktapam_sudo_commands_bundle" "test_acc_sudo_commands_bundle" {
+		name = "%s"
+		structured_commands {
+			command       = "/bin/run.sh"
```
Comment (SajanAlexander-okta) [inline oktapam/resource_sudo_commands_bundle_test.go:49]: there are a lot of other attributes that are on this resource - could we get some better coverage?  doesn't have to be everything.  also, when adding more attributes, please be sure to check them for their presence and correctness within the tests.

File: oktapam/resource_sudo_commands_bundle_test.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+	"github.com/okta/terraform-provider-oktapam/oktapam/utils"
+)
+
+func TestAccResourceSudoCommandsBundle(t *testing.T) {
+	checkTeamApplicable(t, true)
+	resourceName := "oktapam_sudo_commands_bundle.test_acc_sudo_commands_bundle"
+	sudoCommandsBundleName := fmt.Sprintf("test-sudo-commands-bundle-%s", randSeq())
+
+	resource.Test(t, resource.TestCase{
```
Comment (SajanAlexander-okta) [inline oktapam/resource_sudo_commands_bundle_test.go:20]: we generally try and run both a create and update for resources when doing a test.

**Review by SajanAlexander-okta — CHANGES_REQUESTED** [review]

File: go.mod [hunk 4]
```diff
@@ -1,16 +1,17 @@
 module github.com/okta/terraform-provider-oktapam
 
-go 1.21
+go 1.20
 
 require (
-	github.com/atko-pam/pam-sdk-go v1.0.68
+	github.com/atko-pam/pam-sdk-go v1.0.70-0.20240605060748-0f7af3861a5e
```
[the remark this answers is not in the document]
Comment (UlfatSayyed-okta) [reply go.mod:6]: Yes, this is in pending review, have it here to prove it's tested. I can change it once that's merged.

[the remark this answers is not in the document]
Comment (UlfatSayyed-okta) [reply go.mod:6]: pam-sdk PR https://github.com/atko-pam/pam-sdk-go/pull/64 and the spec PR https://github.com/atko-pam/platform/pull/8512

File: oktapam/data_source_sudo_commands_bundles_test.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccSudoCommandsBundlesCheckDestroy(identifier+"-1", identifier+"-2"),
+		Steps: []resource.TestStep{
+			{
+				Config: initConfig,
+			},
+			{
+				Config: fmt.Sprintf("%s\n%s", initConfig, list1Config),
```
[the remark this answers is not in the document]
Comment (UlfatSayyed-okta) [reply oktapam/data_source_sudo_commands_bundles_test.go:34]: Done

File: go.mod (hunk 4 above)
[the remark this answers is not in the document]
Comment (UlfatSayyed-okta) [reply go.mod:6]: Now using  https://github.com/atko-pam/pam-sdk-go/releases/tag/v1.0.72

File: oktapam/client/sudo_command_bundle.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+		}
+	} else if callErr != nil {
+		return nil, callErr
+	}
+
+	return resp, nil
+}
+
+func ListSudoCommandsBundles(ctx context.Context, sdkClient SDKClientWrapper) (*pam.ListSudoCommandBundleResponse, error) {
+	request := sdkClient.SDKClient.SudoCommandsAPI.ListSudoCommandBundles(ctx, sdkClient.Team)
```
[the remark this answers is not in the document]
Comment (UlfatSayyed-okta) [reply oktapam/client/sudo_command_bundle.go:27]: @SajanAlexander-okta - I was looking into adding support for pagination here but I see the generated sudo command bundle request [does not have](https://github.com/atko-pam/pam-sdk-go/blob/master/client/pam/api_sudo_commands.go#L259) the pagination-related fields like [ListSecrets has](https://github.com/atko-pam/pam-sdk-go/blob/master/client/pam/api_secrets.go#L666). Should I manually add it to the sdk or am I missing something?

[the remark this answers is not in the document]
Comment (SajanAlexander-okta) [reply oktapam/client/sudo_command_bundle.go:27]: So things to check: 

 - does the API itself support pagination?
 - does the open api spec define the pagination fields in the response?

**Comment by UlfatSayyed-okta** [comment]
Closing this because created pull/132 with verified commmits

**Pull request closed without merging**